consul/config/agent.hcl

```diff
@@ -0,0 +1,60 @@
+{{ #if cfg.agent.bind_addr }}
+bind_addr = "{{cfg.agent.bind_addr}}"
+{{ else }}
+bind_addr = "{{sys.ip}}"
+{{ /if }}
+
+{{ #if cfg.agent.data_dir }}
+data_dir = "{{cfg.agent.data_dir}}"
+{{ else }}
+data_dir = "{{pkg.svc_data_path}}"
+{{ /if }}
+
+{{ #if cfg.agent.client_addr }}
+client_addr = "{{cfg.agent.client_addr}}"
+{{ else }}
+client_addr = "{{sys.ip}}"
+{{ /if }}
+
+{{ #if cfg.encryption.gossip }}
+encrypt                 = "{{cfg.encryption.gossip_key}}"
+encrypt_verify_incoming = {{cfg.encryption.verify_gossip_incoming}}
+encrypt_verify_outgoing = {{cfg.encryption.verify_gossip_outgoint}}
+{{ /if }}
+
+{{ #if cfg.encryption.tls }}
+ca_file                = "{{cfg.encryption.ca_file}}"
+cert_file              = "{{cfg.encryption.cert_file}}"
+key_file               = "{{cfg.encryption.key_file}}"
+verify_incoming        = {{cfg.encryption.verify_incoming}}
+verify_outgoing        = {{cfg.encryption.verify_outgoing}}
+verify_server_hostname = {{cfg.encryption.verify_server_hostname}}
+{{ /if }}
+
+{{ #if cfg.agent.server }}
```

Comment:
@smacfarlane I feel like this falls into the configuration plan pattern. Its the same binary with differing configuration.

Reply:
@Defilan may have some thoughts on this too.

Reply:
@predominant, do you have an example of the configuration plan pattern that I can use as a model to update this plan?

Reply:
The configuration plan pattern would not require modifying this plan, you would require this plan as a dependency, and add custom configuration yourself. I wrote a decent blog post about it here: https://grahamweldon.com/post/2018/09/habitat-configuration/wrapper-plans/

Reply:
Ah! This is exactly what I ended up doing for my Habitat/Consul demo at ChefConf. Makes sense to me. I can close the PR and maybe document how to do server/client configs using the config plan pattern vs making changes to this plan. Thoughts?

Reply:
Our approach has been to exclude configuration plans from the core-plans repository. Core plans provides binaries and common services. Anything beyond that basic example, and we start to delve into the realm of non-core origins for plans.

If the changes required to achieve the desired functionality are purely configuration, we're leaving that up to the implementing users, rather than adding more plans and configuration complexity to the core-plans repo.

Does that make sense?